pkg/maps/ctmap/ctmap.go

```diff
@@ -550,3 +554,72 @@ func Exists(e CtEndpoint, ipv4, ipv6 bool) bool {
 
 	return result
 }
+
+var cachedGCInterval time.Duration
+
+// GetInterval returns the interval adjusted based on the deletion ratio of the
+// last run
+func GetInterval(mapType bpf.MapType, maxDeleteRatio float64) (interval time.Duration) {
+	if val := option.Config.ConntrackGCInterval; val != time.Duration(0) {
+		interval = val
+		return
+	}
+
+	if interval = cachedGCInterval; interval == time.Duration(0) {
+		interval = defaults.ConntrackGCStartingInterval
+	}
+
+	return calculateInterval(mapType, interval, maxDeleteRatio)
+}
+
+func calculateInterval(mapType bpf.MapType, prevInterval time.Duration, maxDeleteRatio float64) (interval time.Duration) {
+	interval = prevInterval
+
+	if maxDeleteRatio == 0.0 {
+		return
+	}
+
+	switch {
+	case maxDeleteRatio > 0.25:
+		if maxDeleteRatio > 0.9 {
+			maxDeleteRatio = 0.9
+		}
+		// 25%..90% => 1.3x..10x shorter
+		interval = time.Duration(float64(interval) * (1.0 - maxDeleteRatio))
+
+		if interval < defaults.ConntrackGCMinInterval {
+			interval = defaults.ConntrackGCMinInterval
+		}
+
+	case maxDeleteRatio < 0.05:
+		// When less than 5% of entries were deleted, increase the
+		// interval. Do not assume that less than 1% were deleted.
+		if maxDeleteRatio < 0.01 {
+			maxDeleteRatio = 0.01
+		}
+		// 5%..1% => 2x..10x longer
+		interval = time.Duration(float64(interval) * (0.1 / maxDeleteRatio))
+
+		switch mapType {
+		case bpf.MapTypeLRUHash:
+			if interval > defaults.ConntrackGCMaxLRUInterval {
+				interval = defaults.ConntrackGCMaxLRUInterval
+			}
+		default:
+			if interval > defaults.ConntrackGCMaxInterval {
+				interval = defaults.ConntrackGCMaxInterval
+			}
+		}
+	}
+
+	if interval != prevInterval {
```

Comment:
To avoid this condition from being true more times than necessary, round or truncate `interval` to the second, or even, to the minute.

Reply:
Very good point 👍